### ISSUE TITLE
Error in ICLabel feature extraction

### ISSUE THREAD

**Issue by Hamza-Dal** [eeglabdevelopers/eeglab#640]
I am encountering an error while using the ICLabel function in EEGLab. The error occurs within my preprocessing pipeline, which includes resampling, filtering, clean_artifacts, re-referencing, making the data full ranked, performing ICA, and then using ICLabel. The EEG data is in .edf format, and I have already updated EEGLab to the latest version (2023.0) and reinstalled the plugin. 

Relevant files:
[IClabel issue files.zip](https://github.com/sccn/eeglab/files/11894308/IClabel.issue.files.zip)

Below is the error message:
> Error message:
> Index in position 1 exceeds array bounds. Index must not exceed 101.
> Error in signal.internal.resample.uniformResample (line 135)
>    yM=ymat(indM,:);
> Error in resample (line 198)
>            signal.internal.resample.uniformResample(xIn, isDimValSet, Dim, dimIn, numericArgs{:});
> Error in eeg_autocorr_welch (line 48)
>    ac = resample(double(ac'), 100, EEG.srate)';
> Error in ICL_feature_extractor (line 77)
>            autocorr = eeg_autocorr_welch(EEG);
> Error in iclabel (line 52)
> features = ICL_feature_extractor(EEG, flag_autocorr);
> Error in pop_iclabel (line 109)
>    EEG = iclabel(EEG, icversion)

The error occurs on line 171 when calling the pop_iclabel function, which labels independent components. Please find attached the script, sample data and channel layout files. 

Thank you for any guidance you can provide and please let me know if there's any additional information I can provide.

Additional Notes:
>1) Included are channel location files that contain layout information and are used by the pop_chanedit function on line 122. Please modify lines 24 and 25 to the path of the channel locations files.
>2) Enter the path of the attached .edf file as a character variable on line 40. Three sample .edfs are attached. Source for the dataset is the [TUH Corpus ](https://isip.piconepress.com/projects/tuh_eeg/)

Kind regards,
Hamza Imtiaz
PhD Student - Friedman lab
Dalhousie University

**Comment by Hamza-Dal** [comment]
(27/06) Update: I was able to pinpoint the error to the eeg_autocorr_welch.m function. The issue was caused by this line:

`
ac = resample(double(ac'), 100, EEG.srate)';
`

In my case, EEG.srate was 250 and ac was (19x251), a mismatched size for the resample function. 

I replaced the line above within the eeg_autocorr_welch.m function with a manual resampling method, shown below:
```
    desiredSize = EEG.srate;  % Desired output size
    currentSize = size(ac, 2);  % Current size of ac

    % Resample ac to the desired size
    x = linspace(1, currentSize, desiredSize);
    ac_resampled = interp1(1:currentSize, ac', x, 'linear')';

    % Update ac with the resampled matrix
    ac = ac_resampled;
```

It has resolved the issue but I'm not sure whether this was a bug or whether the solution has an effect on the integrity of ICLabel's results.
